src/test/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAccountManagerTest.java

```diff
@@ -125,9 +126,10 @@ private static Stream<Arguments> namesToTest() {
         Arguments.of(false, null), // Null is not valid
         Arguments.of(false, ""), // Empty string is not valid
         Arguments.of(false, " "), // Only whitespace is not valid
-        Arguments.of(false, "a".repeat(256)), // Names exceeding the maximum length of 255 characters are not valid
+        Arguments.of(true, "a".repeat(50)), // The maximum length of 50 characters are valid
+        Arguments.of(false, "a".repeat(51)), // Names exceeding the maximum length of 255 characters are not valid
         // Other special characters are not permitted
-        Arguments.of(false, "Test!Name"),
+        Arguments.of(true, "Test!Name"),
```

Comment:
This is now permitting exclamation marks

Reply:
changed the location of the test. It is now above the comment.